netconn_notjs.go

```diff
@@ -0,0 +1,26 @@
+// +build !js
+
+package websocket
+
+import "net"
+
+type LocalRemoteAddr interface {
+	RemoteAddr() net.Addr
+	LocalAddr() net.Addr
+}
+
+func (c *netConn) RemoteAddr() net.Addr {
+	if ra, ok := c.c.rwc.(LocalRemoteAddr); ok {
```

Comment:
Not sure how this could work? Have you tested this? `c.c.rwc` is not a `net.Conn` and so doesn't have these methods you're checking for.

Reply:
It only works in tests because the tests don't use a real http transport/server, they're using my pipe abstraction. See https://github.com/nhooyr/websocket/blob/master/internal/test/wstest/pipe.go

Reply:
It works fine. Its check the read/writer also RemoteAddr() and LocalAddr() defined.

Reply:
Here is example of something similar to what I did:
https://github.com/golang/go/blob/master/src/io/io.go#L408

Reply:
Have you tested this in production? There's no RemoteAddr or LocalAddr on the bufio.ReadWriter returned to us from the transport.

Reply:
This the echo exmaple.
<img width="1495" alt="Screen Shot 2023-03-08 at 12 10 06 PM" src="https://user-images.githubusercontent.com/3271896/223783888-a5c1e69e-a364-4e9f-83d4-4c959d8459c1.png">

Reply:
Ok now I'm super confused I'll have to look into how that's possible lol